rfcs/053-filtering-by-subject/README.md

```diff
@@ -0,0 +1,101 @@
+# RFC 053: Filtering by contributor, genre, and subject
+
+For per-subject and per-person pages, we want to filter for images and works that match a given subject/person.
+For example:
+
+*   Works by Florence Nightingale
+*   Works about Charles Darwin
+*   Images about mental health
+
+This turns out to be non-trivial, so this RFC describes how we'll get there.
+
+Strictly speaking we only need subjects and contributors for this work, but genres are so similar we should treat them in the same way.
+
+## Requirements
+
+1.  On per-concept pages, there's a sample of matching images/works.
+
+2.  On per-concept pages, there's a link to a filtered search for the given concept, directly below the sample results.
+
+3.  On work pages, the list of subjects/contributors/genres to link to filtered searches for each subject.
+
+4.  In the works API, there are filters and aggregations for subject/contributor/genre.
+
+5.  Given a single work in the works API, there should be an obvious way to construct a filter URL for works with the same subjects/contributors/genres as this work.
+
+6.  The catalogues are the source of truth for subject identifiers.
+    We can find equivalent identifiers, but we can't pick them from scratch.
+
+    e.g. If a Sierra record has a subject tagged with an LCSH identifier, we can find the Wikidata subject with that identifier.
+    If a Sierra record has a subject with no identifier, we can't choose an identifier, even if we could find a Wikidata subject with a matching label.
```

Comment:
These requirements might seem oddly specific; I added them as a way to negate some of the more wild ideas I had. (e.g. have "shadow identifiers" for unidentified concepts, but then it's impossible for anyone to work out how to filter!)

They did lead me towards the suggestion in "An idea".

Reply:
I agree that if something is unidentified, we shouldn't make up an id. It should remain unidentified and that may mean that we have to treat it differently, eg the difference between identified and unidentified series on works pages. But it is being true to the data. We are more likely to get into a pickle if we don't go with the grain of what we have.